zyte_api/_sync.py

```diff
@@ -0,0 +1,111 @@
+import asyncio
+from tenacity import AsyncRetrying
+from typing import Generator, List, Optional, Union
+
+from aiohttp import ClientSession
+
+from .aio.client import AsyncClient
+from .constants import API_URL
+
+
+class ZyteAPI:
```

Comment:
Do you know if it works in IPython notebook?

Reply:
Will check.

Reply:
TIL `RuntimeError: asyncio.run() cannot be called from a running event loop`

Reply:
Neither sync method works on Jupyter, however the async API works. Maybe it would be best for people to use the async API in Jupyter?

Reply:
Do we have any other options? Is it possible to detect that an asyncio loop is running, and if that's the case, start a thread and do the request there in another loop?

Reply:
I recall this was the reason zyte-autoextract's sync client (https://github.com/zytedata/zyte-autoextract/blob/master/autoextract/sync.py) was implemented using requests, not with this straightforward approach. But maybe there is some workaround possible, e.g. they seem to try to do it here:

https://github.com/langchain-ai/langchain/blob/71d0981f18dd2613261ab72d584b674c0cf180fa/libs/community/langchain_community/document_loaders/async_html.py#L204

Overall - probably yes, we might recommend async in the notebooks or in ipython shell, but when would you use sync method? For production integration async is probably better. CLI scripts, regular Python shell?

Reply:
> CLI scripts, regular Python shell?

Yes.